src/types.ts

```diff
@@ -45,5 +45,5 @@ export interface Pagination {
 
 export interface PaginatedResponse<T> {
     pagination?: Pagination;
-    [key: string]: T[] | Pagination;
+    [key: string]: T[] | Pagination | undefined;
```

Comment:
Would it be better to do `[key: string]?: `?

Reply:
I tried and it wasn't happy about it

```
node_modules/eventbrite/lib/types.d.ts:39:18 - error TS1005: ';' expected.

39     [key: string]?: T[] | Pagination;
                    ~

node_modules/eventbrite/lib/types.d.ts:39:19 - error TS1131: Property or signature expected.

39     [key: string]?: T[] | Pagination;
                     ~

node_modules/eventbrite/lib/types.d.ts:39:21 - error TS1131: Property or signature expected.

39     [key: string]?: T[] | Pagination;
                       ~

node_modules/eventbrite/lib/types.d.ts:39:23 - error TS1011: An element access expression should take an argument.

39     [key: string]?: T[] | Pagination;
                         

node_modules/eventbrite/lib/types.d.ts:40:1 - error TS1128: Declaration or statement expected.

40 }
   ~
```

Reply:
Sorry for chiming in so later on this. I don't think you can make a property indexer optional, as @danielepolencic already showed.

My question is why is it:

```
[key: string] T[] | Pagination
```

Why the `| Pagination`? Why would it be a `Pagination`. Can it just be:

```
export interface PaginatedResponse<T> {
    pagination: Pagination;
    [key: string]: T[];
}
```

Will that fix it by removing the ambiguity?

Reply:
```
node_modules/eventbrite/lib/types.d.ts:38:5 - error TS2411: Property 'pagination' of type 'Pagination' is not assignable to string index type 'T[]'.

38     pagination: Pagination;
       ~~~~~~~~~~
```

 It doesn't work.